File: my-folder/0040-combination-sum-ii/solution.cpp
class Solution {
public:
    void recursion(vector<int> &candidates,vector<vector<int>>& ans,vector<int>temp,int i,int sum, int target){
        if(sum==target){
            ans.push_back(temp);
            return;
        }
        if(sum>target || i==candidates.size()) return;
        for(int j=i;j<candidates.size();j++){
            if (j != i && candidates[j] == candidates[j - 1])continue;
            temp.push_back(candidates[j]);
            recursion(candidates,ans,temp,j+1,sum+candidates[j],target);
            temp.pop_back();
        }
    }
    vector<vector<int>> combinationSum2(vector<int>& candidates, int target) {
        vector<vector<int>> ans;
        sort(candidates.begin(),candidates.end());
        recursion(candidates,ans,{},0,0,target);
        return ans;

    }
};

File: my-folder/0040-combination-sum-ii/solution.java
class Solution {
    void getList(int[] candidates,int target, List<List<Integer>> ans,List<Integer> temp,int ind){
        if(target==0){
            ans.add(new ArrayList<>(temp));
            return;
        }
        for(int i = ind;i<candidates.length;i++){
            if(i>ind && candidates[i]==candidates[i-1]) continue;
            if(candidates[i]>target) break;
            temp.add(candidates[i]);
            getList(candidates,target-candidates[i],ans,temp,i+1);
            temp.remove(temp.size()-1);

        }
        
    }
    public List<List<Integer>> combinationSum2(int[] candidates, int target) {
        Arrays.sort(candidates);
        int n = candidates.length;
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        getList(candidates,target,ans,temp,0);
        return ans;
        
    }
}
